src/sentry/static/sentry/app/components/createAlertButton.tsx

```diff
@@ -0,0 +1,266 @@
+        </React.Fragment>
+      )}
+      {totalErrors > 1 && (
+        <React.Fragment>
+          {t(
+            'The world is a cruel and unforgiving place and that button didn’t work because:'
+          )}
+          <StyledUnorderedList>
+            {hasProjectError && <li>{t('One project must be selected')}</li>}
+            {hasEnvironmentError && <li>{t('One or All Environments is required')}</li>}
+            {hasEventTypeError && (
+              <li>
+                {tct(
+                  'Either [error:event.type:error] or [transaction:event.type:transaction] is required',
+                  {
+                    error: <StyledCode />,
+                    transaction: <StyledCode />,
+                  }
+                )}
+              </li>
+            )}
+            {hasYAxisError && (
+              <li>
+                {tct(
+                  '[yAxis] is not supported by alerts just yet. Select a different metric below and try again.',
+                  {
+                    yAxis: <StyledCode>{eventView.getYAxis()}</StyledCode>,
+                  }
+                )}
+              </li>
+            )}
+          </StyledUnorderedList>
+        </React.Fragment>
+      )}
+      <StyledCloseButton
+        icon={<IconClose color="yellow400" size="sm" isCircled />}
+        aria-label={t('Close')}
+        size="zero"
+        onClick={onClose}
+        borderless
+      />
+    </StyledAlert>
+  );
+}
+
+type Props = React.ComponentProps<typeof Button> & {
+  className?: string;
+  projects: Project[];
+  /**
+   * Discover query used to create the alert
+   */
+  eventView: EventView;
+  organization: Organization;
+  /**
+   * Called when the eventView does not meet the requirements of alert rules
+   */
+  onIncompatibleQuery: (incompatibleQuery: React.ReactNode) => void;
+  /**
+   * Called when the alert close button is clicked
+   */
+  onClose: () => void;
+  /**
+   * Called when the user is redirected to the alert builder
+   */
+  onSuccess: () => void;
+};
+
+function incompatibleYAxis(eventView: EventView): boolean {
+  const column = explodeFieldString(eventView.getYAxis());
+  if (column.kind === 'field') {
+    return true;
+  }
+
+  const eventTypeMatch = eventView.query.match(/event\.type:(transaction|error)/);
+  if (!eventTypeMatch) {
+    return false;
+  }
+
+  const dataset = eventTypeMatch[1];
+  const yAxisConfig = dataset === 'error' ? errorFieldConfig : transactionFieldConfig;
+
+  const invalidFunction = !yAxisConfig.aggregations.includes(column.function[0]);
+  // Allow empty parameters
+  const invalidParameter = !['', ...yAxisConfig.fields].includes(column.function[1]);
+
+  return invalidFunction || invalidParameter;
+}
+
+/**
+ * Provide a button that can create an alert from an event view.
+ * Emits incompatible query issues on click
+ */
+function CreateAlertButton({
+  projects,
+  eventView,
+  organization,
+  onIncompatibleQuery,
+  onSuccess,
+  onClose,
```

Comment:
Should it be `onCloseAlert`?

How about if instead of `onIncompatibleQuery` just giving the alert node as an argument, it gives you a `renderIncompatibleAlert` function, that you can then pass your onClose handler to?

Reply:
@evanh really good idea, i changed it to be a function that you pass the onclose handler to.